buildifier/buildifier.go

```diff
@@ -253,14 +256,33 @@ var diff *differ.Differ
 
 // processFile processes a single file containing data.
 // It has been read from filename and should be written back if fixing.
-func processFile(filename string, data []byte) {
+func processFile(filename string, data []byte, inputType string) {
 	defer func() {
 		if err := recover(); err != nil {
 			fmt.Fprintf(os.Stderr, "buildifier: %s: internal error: %v\n", filename, err)
 			exitCode = 3
 		}
 	}()
 
+	defaultMode := tables.FormattingMode
```

Comment:
factor out to keep this function shorter -- pass back a cleanup
func setFormattingMode(inputType string) (cleanup func())

here:
cleanup := setFormattingMode(inputType)
defer cleanup()

Reply:
Like this?